dowhy/gcm/confidence_intervals_cms.py

```diff
@@ -0,0 +1,70 @@
+"""This module provides functionality to estimate confidence intervals via bootstrapping the fitting and sampling.
+
+Functions in this module should be considered experimental, meaning there might be breaking API changes in the future.
+"""
+
+from functools import partial
+from typing import Union, Callable, Any, Dict
+
+import numpy as np
+import pandas as pd
+
+from dowhy.gcm.cms import ProbabilisticCausalModel, InvertibleStructuralCausalModel, StructuralCausalModel
+from dowhy.gcm.fitting_sampling import fit
+
+# A convenience function when computing confidence intervals specifically for causal queries. This function
+# bootstraps only sampling. Use bootstrap_training_and_sampling when you want to bootstrap training too.
+#
+# **Example usage:**
+#
+#     >>> gcm.fit(causal_model, data)
+#     >>> scores_median, scores_intervals = gcm.confidence_intervals(
+#     >>>     gcm.bootstrap_sampling(gcm.estimate_anomaly_scores, causal_model, target_node='Y'))
+bootstrap_sampling = partial
```

Comment:
very nice use of `partial`!  But for a reader's understanding, it may be good to describe what exactly happens in the call to `confidence_intervals` on line 21-22. Why is the `bootstrap_sampling` call required? Why not directly pass `estimate_anomaly_scores`?

Reply:
Does the new explanation help to clarify this?

We are also planning to add a dedicated user guide for the confidence intervals to the documentation soon.